backend/wayland/seat.c

```diff
@@ -276,6 +276,95 @@ static struct wl_keyboard_listener keyboard_listener = {
 	.repeat_info = keyboard_handle_repeat_info
 };
 
+static void touch_coordinates_to_absolute(struct wlr_wl_input_device *device,
+		wl_fixed_t x, wl_fixed_t y, double *sx, double *sy) {
+	// TODO: each output needs its own touch
+	struct wlr_wl_output *output, *tmp;
+	wl_list_for_each_safe(output, tmp, &device->backend->outputs, link) {
```

Comment:
I think `wl_list_for_each` is sufficient here since we don't modify list:
```suggestion
	struct wlr_wl_output *output;
	wl_list_for_each(output, &device->backend->outputs, link) {
```